pkg/flightsql/flightsql.go

```diff
@@ -24,12 +24,9 @@ var (
 	_ backend.CallResourceHandler   = (*FlightSQLDatasource)(nil)
 )
 
-const mdBucketName = "bucket-name"
-
 type config struct {
 	Addr     string            `json:"host"`
-	Database string            `json:"database"`
-	Metadata map[string]string `json:"metadata"`
+	Metadata []map[string]string `json:"metadata"`
```

Comment:
Any reason this needs to be a slice of maps?

Reply:
the frontend is building an array of objects like so: `[{bucket-name: 'sample_data'},{bucket-id: '12345678'}]` so this enables me to unmarshal it and then set the keys

Reply:
Okay, my brain took a journey with this one. I was thinking we would fold it down to a single map to validate. 

However, I see now there is value in saving it this way or as a list of tuples, because you maintain the original order of entry. Good UX.

I haven't checked yet, but do we do any duplicate key validation in the UI or should we do that server-side?

Reply:
I can add some UI side, I need to do that in some other areas as well where you already have it in the backend i'll add an issue